### PR TITLE
Fix base64 encoding of k8s templated values

### PR DESCRIPTION
**Pull request by danielvdende**
## Summary:
Until now, Takeoff made some assumptions regarding base64 encoding,
which in some cases led to awkward situations. The best example was
that in order to render a value from Azure Keyvault into the value of a
Kubernetes secret, the value in the Azure Keyvault would need to be
base64 encoded. 

This commit makes 2 general assumptions:
- template values that are fetched from the keyvault are to be base64
encoded, as they are secret.
- custom values that are provided via Takeoff config are not to be
base64 encoded, as they are already in plain text in deployment.yml

In order to allow maximum flexibility, a set of custom jinja2 filters is also
added, that allows users to deviate from the assumptions mentioned
above, by enforcing base64 encoding or decoding in the kubernetes
configuration. This option should be used with a lot of care, as it
could result in leaking secrets.

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] There is no commented out code in this PR.
  - [x] There are no new TODOs in this PR

If exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated

### CODE REVIEW

File: takeoff/util.py [hunk 1]
```diff
@@ -79,6 +80,53 @@ def b64_encode(s: str) -> str:
...
+jinja2.filters.FILTERS["b64_encode"] = b64_encode
+jinja2.filters.FILTERS["b64_decode"] = b64_decode
+
+
+def is_base64(target: Union[str, bytes]) -> bool:
+    """
+    Courtesy of
+    https://stackoverflow.com/questions/12315398/verify-is-a-string-is-encoded-in-base64-python
+
+    Args:
```
Comment (timvancann) [inline takeoff/util.py]: Missing docstrings here.

File: takeoff/azure/deploy_to_kubernetes.py [hunk 2]
```diff
@@ -241,12 +242,11 @@ def _create_image_pull_secret(self, application_name: str) -> str:
         return self._render_and_write_kubernetes_config(
             kubernetes_config_path=pull_secrets_yaml,
             application_name=application_name,
-            secrets=[
-                Secret("pull_secret", self._get_docker_registry_secret()),
-                Secret("namespace", self.config["image_pull_secret"]["namespace"]),
-                Secret("secret_name", self.config["image_pull_secret"]["secret_name"]),
-            ],
-            custom_values={},
+            secrets=[Secret("pull_secret", self._get_docker_registry_secret())],
```
Comment (timvancann) [inline takeoff/azure/deploy_to_kubernetes.py]: What is the logic for choosing `secrets` or `custom_values`?

Comment (danielvdende) [reply takeoff/azure/deploy_to_kubernetes.py]: So the base assumptions I made are the following: (copied this from the docs update)
```
- Any values that are inserted into your Kubernetes template that originate from the *Keyvault* are assumed to be confidential. As such, Takeoff will ensure that these values
are base64 encoded, as they should only be used in Kubernetes secrets.
- Any values that are inserted into your Kubernetes template via the *custom values* support (i.e. that are supplied in Takeoff's deployment.yml) are assumed to __not__ be confidential. These
values will therefore be inserted into the template in plain text. ``` 

TL;DR: `secrets` are base64 encoded by default, `custom_values` are not. You can deviate from this using the jinja2 filters if you like, but it seemed like a good baseline

File: takeoff/util.py (hunk 1 above)
Comment (danielvdende) [reply takeoff/util.py]: Fixed

File: takeoff/azure/deploy_to_kubernetes.py (hunk 2 above)
Comment (timvancann) [reply takeoff/azure/deploy_to_kubernetes.py]: Aight, that TL;DR actually makes more sense than the documentation to me xD. Also, how about the context values? Those are not "custom" per se, they should be secrets, but are not in the secrets variable.

Comment (danielvdende) [reply takeoff/azure/deploy_to_kubernetes.py]: Good point, I just added that brief TL;DR to the docs, just to clarify a bit more.

The context values are handled differently. Those are rendered into the kubernetes config [here](https://github.com/schipholgroup/takeoff/pull/28/files#diff-d2fd72adfbeee57a0dfa155f0eaaac49R198) and so are also base64 encoded by default
